lib/collection/src/lib.rs

```diff
@@ -114,13 +122,14 @@ impl Collection {
             let shard_path = shard_path(path, shard_id);
             shards.insert(
                 shard_id,
-                Local(LocalShard::load(shard_id, id.clone(), &shard_path, &config).await),
+                Shard::Local(LocalShard::load(shard_id, id.clone(), &shard_path, &config).await),
             );
             ring.add(shard_id);
         }
         Self {
             shards,
             ring,
+            config,
```

Comment:
`CollectionConfig` might be changed during the https://github.com/qdrant/qdrant/blob/11a6e004c17cf9ede50be7a87e044e224d00d24b/lib/storage/src/content_manager/toc.rs#L236
in shards

Keeping the copy of the config in `Collection` will cause inconsistency if not maintained explicitly.

Reply:
Yes, I agree. That's why I added it to state. I have not yet integrated the actual update logic into the consensus, but you can see that it will be included as part of the `CollectionMetaOperations`.

Reply:
It is planned for #399

Reply:
Got it. So I assume we can merge it as-is, cause the `state` is not used anywhere

Reply:
Yes, ok